src/main/kotlin/ecommerce/exception/GlobalExceptionHandler.kt

```diff
@@ -0,0 +1,36 @@
+package ecommerce.exception
+
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.MethodArgumentNotValidException
+import org.springframework.web.bind.annotation.ControllerAdvice
+import org.springframework.web.bind.annotation.ExceptionHandler
+
+@ControllerAdvice
+class GlobalExceptionHandler {
+    @ExceptionHandler(MethodArgumentNotValidException::class)
+    fun handleConstraintViolations(e: MethodArgumentNotValidException): ResponseEntity<String> {
+        val errors = e.bindingResult.fieldErrors.joinToString("; ") { "${it.field}: ${it.defaultMessage}" }
+        return ResponseEntity.badRequest().body(errors)
+    }
+
+    @ExceptionHandler(IllegalArgumentException::class)
+    fun handleIllegalArgument(e: IllegalArgumentException): ResponseEntity<String> {
+        return ResponseEntity.badRequest().body(e.message)
+    }
+
+    @ExceptionHandler(NotFoundException::class)
+    fun handleNotFound(e: NotFoundException): ResponseEntity<String> {
+        return ResponseEntity.status(HttpStatus.NOT_FOUND).body(e.message)
+    }
+
+    @ExceptionHandler(UnauthorizedException::class)
+    fun handleUnauthorized(e: UnauthorizedException): ResponseEntity<String> {
+        return ResponseEntity.status(HttpStatus.UNAUTHORIZED).body(e.message)
+    }
+
+    @ExceptionHandler(InternalServerErrorException::class)
+    fun handleInternalError(e: InternalServerErrorException): ResponseEntity<String> {
+        return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(e.message)
+    }
+}
```

Comment:
How are all the other unexpected exceptions being handled?
Wouldn't it be better to have an additional handler to manage them in one place? 😉